### ISSUE TITLE
Drag and Drop onto new MacDown window should open the file for editting

### ISSUE THREAD

**Issue by harrycis** [MacDownApp/macdown#749]
When people drag and drop a file onto a new MacDown window, say the one immediately after MacDown is launched, the intention is more likely to open the file for editing instead of getting the file path.

**Comment by franklinyu** [comment]
It depends. For example some user drop images onto the window to insert; others may also want to links to other Markdown files, especially when editing `README.md` for GitHub.

The feature you mentioned makes sense for an opt-in one, though.

**Comment by Jmuccigr** [comment]
Hmmm, if you're dropping onto an open document window, I would expect that the result would be inside that open document.

In particular, if it's a text file, why not assume the user is trying to add the text of the file to the current document? If it's anything else, it would be the path.

Or so I would expect.

If you want to open an existing document, drop it onto the dock icon (which I think is how it works now).

**Comment by niklasberglund** [comment]
Also drag and drop to open is a bit confusing as drag and dropping images of other file types such as images should append it's path.

I think support for opening in the window by dropping with the alt key pressed would be nice. And a preference to flip the behaviour(open by default, append path with alt pressed)

**Comment by Jmuccigr** [comment]
Dragging onto an open document window should never open the dragged file, IMO. To do that, drop the file onto the MacDown icon in the dock, as is normal for other apps as well.

My suggestion:

- If file is text file, drop into window inserts text of the file. (I realize there are a lot of text file types to worry about.)
- Otherwise drop inserts path.

Option-drop causes dropped text files to have their path inserted. There's no need to worry about inserting the contents of anything but a text file, right?

FWIW, BBEdit inserts the contents of dropped files. It does nothing for files for which that doesn't make sense, like image files or PDFs.

**Comment by niklasberglund** [comment]
I think drag and drop onto window to open in new tab of that window would make sense. Wouldn't make that much sense to close the already open document of that window and open the dropped one instead?

**Comment by Jmuccigr** [comment]
Dropping on an open window should put that file into the open document in some way. If you want to open a file with an app, drag the file onto the app's icon in the dock. That's a long-standing convention.

**Comment by niklasberglund** [comment]
@Jmuccigr we agree 100% on this.

IMHO it would be nice if it's possible as an **opt-in** or by pressing the **option key** to open in a new tab in the dropped onto window. That's just an **idea** of behaviour I think would make sense.

**Comment by uranusjr** [comment]
Doesn’t drag-and-dropping onto the *icon* already serves this functionality? Why would you need this if a more accessible target is already available?

**Comment by harrycis** [comment]
@uranusjr Drag and Drop into a window is much more natural user interaction than dropping into the icon. Users are trained to drag and drop to windows instead of icon.

**Comment by Jmuccigr** [comment]
@harrycis I have to disagree. Mac users should be getting trained to drop files on Dock icons in order to have them opened in an app. If you drop a text file into a text-editing window, the file contents should dumped into the open window, not be opened separately. Can't test right now, but at least BBEdit does it this way.

**Comment by franklinyu** [comment]
@harrycis Is there any application that "drag and drop into a window" opens the file in a new window? So that we can feel the behavior.

**Comment by Jmuccigr** [comment]
I just tried with Word and it inserts the text from the dropped file into the open window. Bean inserts the file icon(!) and Pages does nothing.
